pretyping/evarconv.ml

```diff
@@ -1022,10 +1073,11 @@ and evar_eqappr_x ?(rhs_is_already_stuck = false) flags env evd pbty
           in
             (match res with
             | Some (f2,args2) ->
-              evar_eqappr_x flags env evd pbty appr1 (f2,Stack.append_app args2 sk2)
+              evar_eqappr_x flags env evd pbty parents1 parents2 keys1 keys2 None appr1 (f2,Stack.append_app args2 sk2)
             | None -> UnifFailure (evd,NotSameHead))
 
         | _, _ ->
+        let no_cs1 = ref false in
```

Comment:
I see why you use a ref to make f2 and f3 pass extra data, and I see no better way. But it deserves a comment.